docker/Dockerfile

```diff
@@ -33,8 +33,8 @@ WORKDIR /app
 
 # Install Pontoon Python requirements
 COPY requirements/* /app/requirements/
-RUN pip install -U 'pip>=8' && \
-    pip install --no-cache-dir --require-hashes -r requirements/default.txt -r requirements/dev.txt -r requirements/test.txt -r requirements/lint.txt
+RUN uv pip install -U 'pip>=8' && \
```

Comment:
Where is that uv coming from? We need to install it.

Reply:
That's a good question. How was this working locally? 🤔

Reply:
I don't see pip being install via apt-get above. Maybe it's already included in the base image?

Reply:
Yeah:

```
Sideshow:pontoon mathjazz$ docker run --rm python:3.11-bookworm pip --version
Unable to find image 'python:3.11-bookworm' locally
3.11-bookworm: Pulling from library/python
56c9b9253ff9: Already exists
364d19f59f69: Already exists
843b1d832182: Already exists
a348c2a8d946: Already exists
353de11681b2: Already exists
29b316cac89e: Already exists
e1860d8ed6fa: Already exists
Digest: sha256:157a371e60389919fe4a72dff71ce86eaa5234f59114c23b0b346d0d02c74d39
Status: Downloaded newer image for python:3.11-bookworm
pip 24.0 from /usr/local/lib/python3.11/site-packages/pip (python 3.11)
```

Reply:
But this is running `uv pip`, not `pip`.

Trying to figure this out, but it's less trivial than I expected (I can install `uv` and it works, but then it's not available when running `make requirements`, I think because of permissions)

Reply:
I meant that `pip` is installed in Docker automatically with the base image we're using, so the old `RUN pip install...` command just worked.

You probably have `uv` installed locally, but not in Docker?

Maybe this helps?
https://github.com/astral-sh/uv?tab=readme-ov-file#installation

Reply:
My comment `How was this working locally? 🤔` was about running the code in this PR (e.g. `uv pip install`) within the Docker image.

When I tried to run `make build` this morning on a clean system (all builds/images removed), it failed because `uv` was not found, but I'm convinced I tested thoroughly this code before. AFAIK, there's no easy way to install `uv` in the Docker build (no SSH access, `make shell` is not root and I didn't find a quick way to run commands as root).